pkg/scheduler/scheduler.go

```diff
@@ -285,36 +286,49 @@ func findFlavorForResource(
 	wUsed map[string]int64,
 	resourceFlavors map[string]*kueue.ResourceFlavor,
 	cq *cache.ClusterQueue,
-	spec *corev1.PodSpec) (string, int64) {
+	spec *corev1.PodSpec) (string, int64, []error) {
+	var errs []error
+
+	if _, exists := cq.RequestableResources[name]; !exists {
+		errs = append(errs, fmt.Errorf("resource %s not found", name))
+		return "", 0, errs
+	}
+
 	// We will only check against the flavors' labels for the resource.
 	selector := flavorSelector(spec, cq.LabelKeys[name])
 	for _, flvLimit := range cq.RequestableResources[name] {
 		flavor, exist := resourceFlavors[flvLimit.Name]
 		if !exist {
-			log.Error(nil, "Flavor not found", "Flavor", flvLimit.Name)
+			// TODO: if "flavor not found" error emitted too frequently, we'd like to aggregate them into one error.
```

Comment:
I think this should be fine. Most likely different resources would use different flavors.